src/Kafka/ConsumerGroups/components/ConsumerGroupResetOffset.tsx

```diff
@@ -0,0 +1,274 @@
+  isDisconnected: boolean;
+  groupId: string;
+  topics: string[];
+  selectedTopic: string;
+  customOffsetValue: string;
+  setcustomOffsetValue: (customOffsetValue: string) => void;
+  consumers: ConsumerRow[];
+  isSelected: boolean;
+  onClickClose: () => void;
+  onClickResetOffset: () => void;
+  onChangeTopic: (value: string) => void;
+  selectedOffset: OffsetValue;
+  onChangeOffsetValue: (value: OffsetValue) => void;
+  confirmCheckboxChecked: boolean;
+  onConfirmationChange: (value: boolean) => void;
+  SelectAllConsumer: (consumer: ConsumerRow[]) => void;
+  onSelectAll: (isSelected: boolean) => void;
+  onSelectRow: (index: number, isSelected: boolean) => void;
+};
+
+export const ConsumerGroupResetOffset: FunctionComponent<
+  ConsumerGroupResetOffsetProps
+> = ({
+  isDisconnected,
+  groupId,
+  topics,
+  customOffsetValue,
+  setcustomOffsetValue,
+  consumers,
+  onClickClose,
+  isModalOpen,
+  onClickResetOffset,
+  selectedTopic,
+  onChangeTopic,
+  selectedOffset,
+  onChangeOffsetValue,
+  confirmCheckboxChecked,
+  onConfirmationChange,
+  isSelected,
+  onSelectAll,
+  onSelectRow,
+}) => {
+  const { t } = useTranslation(["kafka"]);
+
+  const isResetOffsetDisabled = (): boolean => {
+    return (
+      !selectedTopic ||
+      !selectedOffset ||
+      !confirmCheckboxChecked ||
+      !isDisconnected ||
+      consumers.filter(({ selected }) => selected === true).length === 0
+    );
+  };
+
+  const tableColumns = {
+    partition: t("consumerGroup.partition"),
+    clientId: `${t("consumerGroup.client_id")} + ${t(
+      "consumerGroup.member_id"
+    )}`,
+    current_offset: t("consumerGroup.current_offset"),
+    log_end_offset: t("consumerGroup.log_end_offset"),
+    offset_lag: t("consumerGroup.offset_lag"),
+    new_offset: t("consumerGroup.new_offset"),
+  };
+
+  return (
+    <Modal
+      variant={ModalVariant.large}
+      isOpen={isModalOpen}
+      aria-label={t("consumerGroup.reset_offset_modal_label")}
+      title={t("consumerGroup.reset_offset")}
+      showClose={true}
+      aria-describedby="modal-message"
+      actions={[
+        <Button
+          variant="danger"
+          key={1}
+          isDisabled={isResetOffsetDisabled()}
+          onClick={onClickResetOffset}
+        >
+          {t("consumerGroup.reset_offset")}
+        </Button>,
+        <Button variant="link" key={2} onClick={onClickClose}>
+          {t("common:cancel")}
+        </Button>,
+      ]}
+    >
+      <Stack hasGutter>
+        <StackItem>
+          <Form isHorizontal>
+            <FormGroup
+              className="mas--ConsumerGroupResetOffset-form-group--readonly"
+              label={t("consumerGroup.reset_offset_consumer_group_label")}
+              fieldId="horizontal-form-name"
+            >
+              <TextInput
+                isReadOnly
+                type="text"
+                aria-label={t(
+                  "consumerGroup.reset_offset_consumer_group_label"
+                )}
```

Comment:
Since a text input can use the native html `<label for="">`/`<input id="">` association for providing an accessible name, this is preferred over using `aria-label`. 

To use this method, you can make these changes:
- assign a unique `id` to this `<TextInput>` component )
- use the same value you give the `id` as the value for `fieldID` above
- remove the `aria-label` here